float32 distance
float32 angle
int32 label
